// react-styleguidist/lib/client/rsg-components/Props/renderExtra.js
import "core-js/modules/es.symbol";
import "core-js/modules/es.symbol.description";
import "core-js/modules/es.array.map";
import "core-js/modules/es.function.name";
import "core-js/modules/es.object.keys";
import React from 'react';
import Group from 'react-group';
import Type from 'rsg-components/Type';
import Code from 'rsg-components/Code';
import Name from 'rsg-components/Name';
import Markdown from 'rsg-components/Markdown';
import { unquote, getType, showSpaces } from './util';
import renderDefault from './renderDefault';
import { renderType } from './renderType';

function renderEnum(prop) {
  var type = getType(prop);

  if (!type) {
    return undefined;
  }

  if (!Array.isArray(type.value)) {
    return React.createElement("span", null, type.value);
  }

  var values = type.value.map(function (_ref) {
    var value = _ref.value;
    return React.createElement(Code, {
      key: value
    }, showSpaces(unquote(value)));
  });
  return React.createElement("span", null, "One of: ", React.createElement(Group, {
    separator: ", "
  }, values));
}

function renderUnion(prop) {
  var type = getType(prop);

  if (!type) {
    return undefined;
  }

  if (!Array.isArray(type.value)) {
    return React.createElement("span", null, type.value);
  }

  var values = type.value.map(function (value, index) {
    return React.createElement(Type, {
      key: value.name + "-" + index
    }, renderType(value));
  });
  return React.createElement("span", null, "One of type: ", React.createElement(Group, {
    separator: ", "
  }, values));
}

function renderShape(props) {
  return Object.keys(props).map(function (name) {
    var prop = props[name];
    var defaultValue = renderDefault(prop);
    var description = prop.description;
    return React.createElement("div", {
      key: name
    }, React.createElement(Name, null, name), ': ', React.createElement(Type, null, renderType(prop)), defaultValue && ' — ', defaultValue, description && ' — ', description && React.createElement(Markdown, {
      text: description,
      inline: true
    }));
  });
}

export default function renderExtra(prop) {
  var type = getType(prop);

  if (!prop.type || !type) {
    return null;
  }

  switch (type.name) {
    case 'enum':
      return renderEnum(prop);

    case 'union':
      return renderUnion(prop);

    case 'shape':
      return renderShape(prop.type.value);

    case 'arrayOf':
      if (type.value.name === 'shape') {
        return renderShape(prop.type.value.value);
      }

      return null;

    case 'objectOf':
      if (type.value.name === 'shape') {
        return renderShape(prop.type.value.value);
      }

      return null;

    default:
      return null;
  }
}
